source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -328,16 +330,20 @@ void HystrixSink::flush(Stats::Source& source) {
   // Save a map of the relevant histograms per cluster in a convenient format.
   std::unordered_map<std::string, QuantileLatencyMap> time_histograms;
   for (const Stats::ParentHistogramSharedPtr& histogram : source.cachedHistograms()) {
-    if (histogram->tagExtractedStatName() == cluster_upstream_rq_time_.statName()) {
-      // TODO(mrice32): add an Envoy utility function to look up and return a tag for a metric.
-      auto tags = histogram->tags();
-      auto it = std::find_if(tags.begin(), tags.end(), [](const Stats::Tag& tag) {
-        return (tag.name_ == Config::TagNames::get().CLUSTER_NAME);
-      });
+    if (histogram->tagExtractedStatName() == cluster_upstream_rq_time_) {
+      std::string value;
+      histogram->iterateTagStatNames(
```

Comment:
I can see this pattern being used a lot. Is it possible to add a utility function that takes a metric, symbol table and StatName tag to look for and returns an abl::optional<StatName> value?

Reply:
Agreed -- that's a great simplification; will do. I'll need to get back to this though as I am not free to code ATM :). One other thought is that it I am debating whether this should be in a utility layer over the virtual class hierarchy around Metric, rather than in the class hierarchy itself, so it doesn't need to be replicated in the mocks.

Reply:
> One other thought is that it I am debating whether this should be in a utility layer over the virtual class hierarchy around Metric, rather than in the class hierarchy itself, so it doesn't need to be replicated in the mocks.

If we make this a final, non-virtual helper method can we avoid needing it to be replicated in mocks?

Reply:
I put the helper function into source/common/stats/utility.(h|cc)